etc/win-ci/crystal.iss

```diff
@@ -0,0 +1,335 @@
+#define MyAppName "Crystal"
+#define VersionFile FileOpen("portable\src\VERSION")
+#define MyAppVersion FileRead(VersionFile)
+#define MyAppVersionNum StringChange(MyAppVersion, "-dev", "")
+#expr FileClose(VersionFile)
+#define MyAppPublisher "Manas Technology Solutions"
+#define MyAppURL "https://crystal-lang.org/"
+#define MyAppExeName "crystal.exe"
+#define MyAppAssocName MyAppName + " Source File"
+#define MyAppAssocExt ".cr"
+#define MyAppAssocKey StringChange(MyAppAssocName, " ", "") + MyAppAssocExt
+
+[Setup]
+AppId={{7C307DDF-447E-46C5-BB3B-47A6F652D7C8}
+AppName={#MyAppName} x86_64-windows-msvc
+AppVersion={#MyAppVersion}
+AppCopyright=Copyright 2012-2023 Manas Technology Solutions
+VersionInfoVersion={#MyAppVersionNum}
+AppPublisher={#MyAppPublisher}
+AppPublisherURL={#MyAppURL}
+AppSupportURL={#MyAppURL}
+AppUpdatesURL={#MyAppURL}
+ArchitecturesAllowed=x64
+ArchitecturesInstallIn64BitMode=x64
+DefaultDirName={autopf}\{#MyAppName}
+OutputBaseFilename=crystal-setup
+LicenseFile=portable\LICENSE.txt
+
+ChangesEnvironment=yes
+ChangesAssociations=yes
+
+DefaultGroupName={#MyAppName} {#MyAppVersion}
+AllowNoIcons=yes
+
+PrivilegesRequired=lowest
+PrivilegesRequiredOverridesAllowed=dialog
+
+Compression=lzma
+SolidCompression=yes
+
+WizardStyle=modern
+WizardImageFile=crystal.bmp
+WizardSmallImageFile=crystal_small.bmp
+DisableWelcomePage=no
+
+SetupMutex={#MyAppName}SetupMutex
+
+[Languages]
+Name: "english"; MessagesFile: "compiler:Default.isl"
+
+[Types]
+Name: "full"; Description: "Full installation"
+Name: "minimal"; Description: "Minimal installation"
+Name: "custom"; Description: "Custom installation"; Flags: iscustom
+
+[Components]
+Name: "main"; Description: "Crystal compiler"; Types: full minimal custom; Flags: fixed
+Name: "shards"; Description: "Shards dependency manager"; Types: full
+Name: "pdb"; Description: "Debug symbols"; Types: full
+Name: "samples"; Description: "Sample programs"; Types: full
+Name: "docs"; Description: "Offline standard library documentation"; Types: full
+
+[Tasks]
+Name: addtopath; Description: "Add Crystal's directory to the &PATH environment variable"; Flags: checkedonce
+Name: association; Description: "{cm:AssocFileExtension,Crystal,.cr}"; Flags: unchecked
+
+[Files]
+Source: "portable\{#MyAppExeName}"; DestDir: "{app}"; Flags: ignoreversion; Components: main
+Source: "portable\lib\*"; DestDir: "{app}\lib"; Flags: ignoreversion; Components: main
+Source: "portable\*.dll"; DestDir: "{app}"; Flags: ignoreversion; Components: main

```

Comment:
Just thinking alout. Dunno if you have given this any thought.
Lib installation could theoretically also be a separate component? Maybe even two (static and dynamic libs)?

Reply:
It's possible to list them as separate mandatory components. I don't think we need to do anything here yet